### PR TITLE
RocksDB as main db. ColumnFamily feature added from rocksdb.

### PR DESCRIPTION
**Pull request by aleksuss**

### CODE REVIEW

File: exonum/benches/block.rs [hunk 1]
```diff
@@ -113,7 +111,7 @@ mod tests {
             }
 
             fn execute(&self, view: &mut Fork) {
-                let mut index = ProofMapIndex::new(vec![1], view);
+                let mut index = ProofMapIndex::new("name", view);
```
Comment (alekseysidorov) [inline exonum/benches/block.rs]: What does this name mean?

File: exonum/src/storage/base_index.rs [hunk 2]
```diff
@@ -52,25 +55,59 @@ pub struct BaseIndexIter<'a, K, V> {
...
+
+    /// Creates a new index representation based on the name, common prefix of its keys
+    /// and storage view.
+    ///
+    /// Storage view can be specified as [`&Snapshot`] or [`&mut Fork`]. In the first case only
+    /// immutable methods are available. In the second case both immutable and mutable methods are
+    /// available.
+    /// [`&Snapshot`]: ../trait.Snapshot.html
+    /// [`&mut Fork`]: ../struct.Fork.html
+    pub fn with_prefix(name: &str, prefix: Vec<u8>, view: T) -> Self {
```
Comment (alekseysidorov) [inline exonum/src/storage/base_index.rs]: Why not `AsRef<str>`?

File: exonum/src/storage/base_index.rs [hunk 3]
```diff
@@ -158,18 +199,18 @@ impl<'a> BaseIndex<&'a mut Fork> {
         K: StorageKey,
     {
         let key = self.prefixed_key(key);
-        self.view.remove(key);
+        self.view.remove(&self.name, key);
     }
 
-    /// Clears the index, removing all entries.
+    /// Clears the index, removing all entries or entries with keys that starts with a prefix.
```
Comment (alekseysidorov) [inline exonum/src/storage/base_index.rs]: Entries or entries? Please rephrase.

File: exonum/src/storage/base_index.rs (hunk 2 above)
Comment (aleksuss) [reply exonum/src/storage/base_index.rs]: For a name I agree but for a prefix i guess it's needless. The prefix will change ownership and won't be used in another place.

File: exonum/src/storage/db.rs [hunk 4]
```diff
@@ -254,66 +260,90 @@ impl Fork {
...
+                    None => changes.remove(&k),
+                };
+            }
         }
         self.logged = false;
     }
 
-    /// Inserts the key-value pair into the fork.
-    pub fn put(&mut self, key: Vec<u8>, value: Vec<u8>) {
+    /// Inserts the key-value pair into the fork with name `name`.
```
Comment (DarkEld3r) [inline exonum/src/storage/db.rs]: ...with **_the_** name `name`. :smile: Maybe paraphrase it as "...into the fork with given name"?

File: exonum/src/storage/db.rs [hunk 5]
```diff
@@ -254,66 +260,90 @@ impl Fork {
...
             ));
         } else {
-            self.changes.insert(key, Change::Put(value));
+            changes.insert(key, Change::Put(value));
         }
     }
 
-    /// Removes the key from the fork.
-    pub fn remove(&mut self, key: Vec<u8>) {
+    /// Removes the key from the fork with name `name`.
```
Comment (DarkEld3r) [inline exonum/src/storage/db.rs]: Same here and below.

File: exonum/src/storage/error.rs [hunk 6]
```diff
@@ -30,7 +30,7 @@ impl Error {
     /// ```
     /// use exonum::storage::Error;
     ///
-    /// let error = Error::new("Oh no!");
+    /// let _error = Error::new("Oh no!");
```
Comment (DarkEld3r) [inline exonum/src/storage/error.rs]: As for me, it isn't ideal to use underscored variables in the examples. Additionally, "unused variable" warnings should be suppressed in the doc-tests, it is just rust-plugin issue (https://github.com/intellij-rust/intellij-rust/issues/1554).

**Pull request merged**